### PR TITLE
Use predefined `compute_∇²!` instead of defining new `calc_∇²!` in `test_matrix_poisson_solver.jl`

### LINKED ISSUE #2638: Is `calc_∇²!` redundant in view of `compute_∇²!`?
Is there a reason that `test_matrix_poisson_solver.jl` defines its own function for computing the laplacian:

https://github.com/CliMA/Oceananigans.jl/blob/95206ae73e10460beebb2ce9d5fc9945523e5724/test/test_matrix_poisson_solver.jl#L8-L15

instead of using the one in `utils_for_runtests.jl`

https://github.com/CliMA/Oceananigans.jl/blob/95206ae73e10460beebb2ce9d5fc9945523e5724/test/utils_for_runtests.jl#L60-L77

I can only note an extra `fill_halo_regions!(ϕ)` line.

cc @simone-silvestri

### PR DESCRIPTION
**Pull request by navidcy**
Closes #2638.

### CODE REVIEW

**Comment by navidcy** [comment]
the matrix solver tests seem ok ;)
https://buildkite.com/clima/oceananigans/builds/8008#0181cb59-5724-4983-8405-c00440044a02

File: test/test_multi_region_poisson_solver.jl [hunk 1]
```diff
@@ -63,11 +54,12 @@ function compute_poisson_weights(grid)
     Az = zeros(N...)
     C  = zeros(grid, N...)
     D  = arch_array(grid.architecture, ones(N...))
-    for i = 1:grid.Nx, j = 1:grid.Ny, k = 1:grid.Nz
+    for k = 1:grid.Nz, j = 1:grid.Ny, i = 1:grid.Nx
```
Comment (francispoulin) [inline test/test_multi_region_poisson_solver.jl:57]: This change is very slight.  Can I ask what the motivation is?

**Review by francispoulin — APPROVED** [review]

File: test/test_multi_region_poisson_solver.jl (hunk 1 above)
Comment (francispoulin) [reply test/test_multi_region_poisson_solver.jl:57]: Thanks!

Comment (navidcy) [reply test/test_multi_region_poisson_solver.jl:57]: Yep, exactly that. Not that it matters so much here, but I just thought we don't want to "create bad habits" :)

Comment (francispoulin) [reply test/test_multi_region_poisson_solver.jl:57]: And thanks to that I learned something.  Much appreciated.

**Pull request merged**